activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -101,7 +101,7 @@ def alias_attribute_method_definition(code_generator, pattern, new_name, old_nam
           super
         else
           define_proxy_call(code_generator, method_name, pattern.proxy_target, parameters, old_name,
```

Comment:
`old_name` is being passed as `call_args` argument of the `define_proxy_call`
https://github.com/rails/rails/blob/e20e8bbbcda46abafaa6e6182cfc21d26f73f2d2/activemodel/lib/active_model/attribute_methods.rb#L406

Perhaps a better fundamental fix would to be change `define_proxy_call` to use `call_args` in the namespace as technically different `call_args` result in different method bodies which means we shouldn't attempt to reuse the method from the cache

It would have fixed our problem by implication. I'll look into changing `define_proxy_call` as an alternative